### DISCUSSION TITLE
Question about part pricing. Can I get the current average part price?

### DISCUSSION THREAD

**Discussion by jacky4566** [inventree/InvenTree#4195] [General]
Hi guys just found this project. Pretty awesome and I hope to contribute somehow. 

Starting to poke around. I work with CIN7 for our small business and while its amazing software, they charge an arm and a leg and never pick up the customer service phone... We are looking for alternatives. 

My first real quick question. Is is possible to see the current average purchase price of a part from what is in stock?

If I buy 2 widgets @ $50, The average widget price is $50.
I then later buy 1 more @ 75. The average widget price is $58.33.

I see the price range stuff but thats not super helpful for the sales staff. they just need to know the current average price of what we have in stock.

**Comment by SchrodingersGat** [comment]
Not currently, although the data is available via the API. It may be quite simple to add this in but there would be a lot of considerations. 

Perhaps you would consider sponsoring such a feature? Or contribute to it yourself?

**Comment by miggland** [comment]
Just to clarify what @SchrodingersGat means by "available in the API", an example with Blue Paint ([Part #89](https://demo.inventree.org/part/89/#)) from the demo server - very simple, and would of course have to be adapted to what you want to calculate (based on what was bought, how many left, how old...?)

Using the `inventree-python` package, it would be something like this. See https://inventree.org/demo for username, password.
```
from inventree.api import InvenTreeAPI
from inventree.stock import StockItem

api = InvenTreeAPI('https://demo.inventree.org',username = 'USERNAME', password = 'PASSWORD')

sti = StockItem.list(api,part=89)
for s in sti:
    print(f"{s['quantity']} bought at {s['purchase_price']} {s['purchase_price_currency']}")
```

Output:
```
10.0 bought at None USD
25.0 bought at 2.550000 EUR
500.0 bought at 10.000000 EUR
```

**Comment by jacky4566** [comment]
Fantastic. I will continue to learn the software and make a list of feature requests. My programming skills are 6/10 so i can make a decent attempt but it may need some clean up. 
Looking forward to seeing where this project goes.
